packages/devtools_app/lib/src/timeline/flutter/timeline_screen.dart

```diff
@@ -333,3 +327,79 @@ class TimelineScreenBodyState extends State<TimelineScreenBody>
         offlineDataJson[TimelineData.traceEventsKey] != null;
   }
 }
+
+class TimelineConfigurationsDialog extends StatelessWidget {
+  const TimelineConfigurationsDialog(this.controller);
+
+  static const dialogWidth = 300.0;
+
+  final TimelineController controller;
+
+  @override
+  Widget build(BuildContext context) {
+    return Dialog(
+      shape: RoundedRectangleBorder(
+          borderRadius: BorderRadius.circular(defaultDialogRadius)),
+      child: Container(
+        width: dialogWidth,
+        padding: const EdgeInsets.all(defaultSpacing),
+        child: Column(
+          mainAxisSize: MainAxisSize.min,
+          mainAxisAlignment: MainAxisAlignment.start,
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: [
+            ...headerInColumn(Theme.of(context).textTheme, 'Recorded Streams'),
+            ..._recordedStreams(),
+          ],
+        ),
+      ),
+    );
+  }
+
+  List<Widget> _recordedStreams() {
+    final settings = <Widget>[];
+    for (final streamName in controller.recordedStreams.keys) {
+      final title = Text(streamName);
+      final checkbox = ValueListenableBuilder(
+        valueListenable: controller.recordedStreams[streamName],
+        builder: (context, enabled, _) {
+          return Checkbox(
+            value: enabled,
+            onChanged: (_) => controller.toggleTimelineStream(streamName),
+          );
+        },
+      );
+      settings.add(
+        Row(
+          mainAxisSize: MainAxisSize.min,
+          mainAxisAlignment: MainAxisAlignment.start,
+          children: [
+            checkbox,
+            title,
+          ],
+        ),
+      );
+    }
+
+    // Special case "Network Traffic" because it is not implemented as a
```

Comment:
If this does have some differences from the POV of the user, we could use a `Divider` widget to visually separate it from the other items.

Reply:
I thought about that. I don't think it does have a difference from the POV of the user. To the user, all the other toggles mean "do I want to see these events in the timeline or not", and this is what the result of the Network toggle will be for the user as well.